core/modules/utils/utils.js

```diff
@@ -1005,4 +1005,100 @@ exports.makeCompareFunction = function(type,options) {
 	return (types[type] || types[options.defaultType] || types.number);
 };
 
+// The character that will substitute for a space in the URL
+var SPACE_SUBSTITUTE = "_";
+
+// The character added to the end to avoid ending with `.`, `?`, `!` or the like
+var TRAILER = "_";
+
+// The character that will separate out the list elements in the URL
+var CONJUNCTION = ";";
+
+// Those of the allowed url characters claimed by TW
+var CLAIMED = [SPACE_SUBSTITUTE, ":", CONJUNCTION];
+
+// Non-alphanumeric characters allowed in a URL fragment
+// More information at https://www.rfc-editor.org/rfc/rfc3986#appendix-A
+var VALID_IN_URL_FRAGMENT = "-._~!$&'()*+,;=:@/?".split("");
+
+// The subset of the pchars we will not percent-encode in permalinks/permaviews
+var SUBSTITUTES = []
+$tw.utils.each (VALID_IN_URL_FRAGMENT, function(c) {if(CLAIMED.indexOf(c) === -1) {SUBSTITUTES.push(c)}});
```

Comment:
This should be multilines, for better readability and there should be no space between `$tw.utils.each(..` and the open brace.